doc/user/spatial_block_parameters.rst

```diff
@@ -0,0 +1,339 @@
+            od: 0.9
+            latticeIDs: [0, 1]
+        void:
+            shape: Circle
+            material: Void
+            Tinput: 20
+            Thot: 20
+            od: 0.819
+            latticeIDs: [0]
+
+Note that we can assign the same component to multiple lattice sites with multiple entries in the ``latticeIDs`` list.
+Also note that we do not need to assign a ``mult`` entry to these components. Their multiplicity will be determined
+based on the number of lattice sites they occupy!
+
+.. seelso::
+
+    The :ref:`LWR tutorial <walkthrough-lwr>` contains additional examples for working with sub-block grids.
+
+Auto grid
+---------
+
+In some cases, you may have an assembly that contains one pin type. The framework provides a mechanism for automatically
+constructing a spatial grid for the block based only on the multiplicity of pin-like components. When constructing
+a block from blueprints, a grid may be added to the block depending on:
+
+1. The existence of an explicitly defined block grid, like in the previously discussed section, and
+2. If the ``autoGenerateBlockGrids`` setting is active.
+
+Should either of these conditions be met, the framework will attempt to add a grid by calling
+:meth:`armi.reactor.blocks.Block.autoCreateSpatialGrids`. However, this behavior is not generalized and only
+implemented on :class:`armi.reactor.blocks.HexBlock`, which makes the following assumptions:
+
+1. You want a corners up hexagonal lattice grid.
+2. The pitch of your hexagonal lattice is determined by :meth:`armi.reactor.blocks.HexBlock.getPinPitch` which
+   may place restrictions on what constitutes a pin.
+3. The number of pins is determined by :meth:`armi.reactor.blocks.HexBlock.getNumPins` which may place similar
+   restrictions on what constitutes a pin.
+
+If the auto grid creation is successful, components with a multiplicity equal to the number of pins will be assigned
+locations on the lattice grid.
+
+.. warning::
+
+    Consider subclassing :class:`~armi.reactor.blocks.HexBlock` with specific pin-like methods and
+    overriding the :meth:`~armi.reactor.blocks.HexBlock.autoCreateSpatialGrids` if you want complete control
+    over this process. Alternatively, use an explicit grid in blueprints.
+
+
+Interacting with spatial data
+=============================
+
+This section will focus on accessing locations of components in the block, locations of specifically
+pins, and examples of some pin data that may be assigned to a block's parameter set.
+
+Component locations
+-------------------
+
+Components that live on a spatial grid have a ``spatialLocator`` attribute to help indicate where that
+component exists in space. If we grab the fuel component from the UO2 block in the
+:ref:`ANL AFCI 177 example <walkthrough-inputs>` we can see where it exists in the block::
+
+    >>> import armi
+    >>> armi.configure()
+    >>> from armi.reactor.flags import Flags
+    >>> r = armi.init(fName="c5g7-settings.yaml").r
```

Comment:
You say you want to use the example reactor `anl-afci-177`, but then you point at the `c5g7` reactor. These are two completely different sets of settings files and blueprints in ARMI's test area.